profiling/src/internal/profile.rs

```diff
@@ -255,6 +258,55 @@ impl Profile {
         const INITIAL_PPROF_BUFFER_SIZE: usize = 32 * 1024;
         let mut encoder = CompressedProtobufSerializer::with_capacity(INITIAL_PPROF_BUFFER_SIZE);
 
+        let timestamped_data = self.observations.timestamped_data();
+        println!("Timestamped data size: {}", timestamped_data.len());
+
+        let mut decompressed_input = lz4_flex::frame::FrameDecoder::new(std::io::Cursor::new(timestamped_data));
+
+        let mut _timeline_samples = 0;
+
+        loop {
+            let stacktrace_id_raw = match decompressed_input.read_u32::<NativeEndian>() {
+                Ok(value) => value,
+                Err(_) => { break; }
+            };
+
+            // let stacktrace_id_raw = decompressed_input.read_u32::<NativeEndian>().unwrap();
+            let labels_id_raw = decompressed_input.read_u32::<NativeEndian>().unwrap();
+            let timestamp_raw = decompressed_input.read_i64::<NativeEndian>().unwrap();
+
+            let mut values: Vec<i64> = Vec::with_capacity(self.sample_types.len());
```

Comment:
Good point. I think that ideally this would move inside the iterator, so we don't need to have two copies of the code for going through each sample.

I tried doing that but found it too hard with my current Rust-foo level >_>

Reply:
I've moved this to be inside the  `TimestampedObservationsIter`. A `Vec` still gets created every time, but with limited lifetime (just each iteration). Suggestions welcome on improvements :)